src/sentry/static/sentry/app/views/settings/project/projectFilters/groupTombstones.jsx

```diff
@@ -62,60 +50,55 @@ const GroupTombstoneRow = createReactClass({
                   'Are you sure you wish to continue?'
               )}
               onConfirm={() => {
-                undiscard(data.id);
+                onUndiscard(data.id);
               }}
             >
               <span className="icon-trash undiscard" />
             </LinkWithConfirmation>
-          </span>
+          </Tooltip>
         </div>
       </li>
     );
-  },
-});
+  }
+}
 
-const GroupTombstones = createReactClass({
-  displayName: 'GroupTombstones',
-
-  propTypes: {
+class GroupTombstones extends AsyncComponent {
+  static propTypes = {
     orgId: PropTypes.string.isRequired,
     projectId: PropTypes.string.isRequired,
     tombstones: PropTypes.array.isRequired,
     tombstoneError: PropTypes.bool.isRequired,
     fetchData: PropTypes.func.isRequired,
-  },
+  };
 
-  mixins: [ApiMixin],
+  getEndpoints() {
+    let {orgId, projectId} = this.props;
+    return [['tombstones', `/projects/${orgId}/${projectId}/tombstones/`]];
+  }
 
-  undiscard(tombstoneId) {
+  handleUndiscard = tombstoneId => {
     let {orgId, projectId} = this.props;
     let path = `/projects/${orgId}/${projectId}/tombstones/${tombstoneId}/`;
     this.api.request(path, {
       method: 'DELETE',
       success: data => {
-        IndicatorStore.add(
-          t('Events similar to these will no longer be filtered'),
-          'success',
-          {duration: 4000}
-        );
+        addSuccessMessage(t('Events similar to these will no longer be filtered'));
       },
       error: () => {
-        IndicatorStore.add(t('We were unable to undiscard this group'), 'error', {
-          duration: 4000,
-        });
+        addErrorMessage(t('We were unable to undiscard this group'));
```

Comment:
This was already here before your patch, but should we not be referring to these as issues (vs groups)? @macqueen 

The empty state below seems to: `'You have no discarded issues'`.